src/cargo/core/resolver/context.rs

```diff
@@ -22,56 +22,26 @@ pub struct ResolverContext {
     pub resolve_features: im_rc::HashMap<PackageId, FeaturesSet, rustc_hash::FxBuildHasher>,
     /// get the package that will be linking to a native library by its links attribute
     pub links: im_rc::HashMap<InternedString, PackageId, rustc_hash::FxBuildHasher>,
-
     /// a way to look up for a package in activations what packages required it
     /// and all of the exact deps that it fulfilled.
     pub parents: Graph<PackageId, im_rc::HashSet<Dependency, rustc_hash::FxBuildHasher>>,
 }
 
+/// By storing activation keys in a `HashMap` we ensure that there is only one
+/// semver compatible version of each crate.
+type Activations = im_rc::HashMap<
+    ActivationKeyNoHash,
+    (Summary, ContextAge),
+    nohash_hasher::BuildNoHashHasher<ActivationKeyNoHash>,
+>;
+
 /// When backtracking it can be useful to know how far back to go.
 /// The `ContextAge` of a `Context` is a monotonically increasing counter of the number
 /// of decisions made to get to this state.
 /// Several structures store the `ContextAge` when it was added,
 /// to be used in `find_candidate` for backtracking.
 pub type ContextAge = usize;
 
-/// Find the activated version of a crate based on the name, source, and semver compatibility.
-/// By storing this in a hash map we ensure that there is only one
-/// semver compatible version of each crate.
-/// This all so stores the `ContextAge`.
-pub type ActivationsKey = (InternedString, SourceId, SemverCompatibility);
```

Comment:
For myself, I'd prefer the moving of these types to be in its own refactor commit